packages/frontend-2/components/project/page/acc/Syncs.vue

```diff
@@ -0,0 +1,536 @@
+import { latestModelsQuery } from '~/lib/projects/graphql/queries'
+import { isArray } from 'lodash-es'
+import {
+  accSyncItemCreateMutation,
+  accSyncItemDeleteMutation,
+  accSyncItemUpdateMutation
+} from '~/lib/acc/graphql/mutations'
+import { projectAccSyncItemsQuery } from '~/lib/acc/graphql/queries'
+import { onProjectAccSyncItemUpdatedSubscription } from '~/lib/acc/graphql/subscriptions'
+import { PauseIcon } from '@heroicons/vue/24/solid'
+import { TrashIcon, PlayIcon } from '@heroicons/vue/24/outline'
+
+const props = defineProps<{
+  projectId: string
+  tokens: AccTokens | undefined
+  isLoggedIn: boolean
+}>()
+
+// TODO ACC: Need to think about data residency from "ACC > Speckle" and warn users accordingly
+
+const step = ref(0)
+
+const showNewSyncDialog = ref(false)
+const { triggerNotification } = useGlobalToast()
+
+const tokens = computed(() => props.tokens)
+const hubs = ref<AccHub[]>([])
+const loadingHubs = ref(false)
+const selectedHub = ref<AccHub | null>(null)
+const selectedHubId = ref<string | null>(null)
+const rootProjectFolderId = ref<string | null>(null)
+const projects = ref<AccProject[]>([])
+const loadingProjects = ref(false)
+const selectedProjectId = ref<string | null>(null)
+const folderContents = ref<AccItem[]>([])
+const selectedFolderContent = ref<AccItem>()
+const loadingFiles = ref(false)
+
+const revitViewName = ref<string>()
+
+const searchText = ref<string>()
+
+const selectedModel = ref<ProjectPageLatestItemsModelItemFragment>()
+
+const latestModelsQueryVariables = computed(
+  (): ProjectLatestModelsPaginationQueryVariables => {
+    const shouldHaveFilter = searchText.value && searchText.value.length > 0
+    return {
+      projectId: props.projectId,
+      filter: shouldHaveFilter
+        ? {
+            search: searchText.value || null
+          }
+        : null
+    }
+  }
+)
+
+const { result: modelsResult } = useQuery(
+  latestModelsQuery,
+  () => latestModelsQueryVariables.value
+)
+
+const models = computed(() => modelsResult.value?.project?.models?.items || [])
+
+const { result: accSyncItemsResult, refetch: refetchAccSyncItems } = useQuery(
+  projectAccSyncItemsQuery,
+  () => ({
+    id: props.projectId
+  })
+)
+
+const accSyncItems = computed(
+  () => accSyncItemsResult.value?.project.accSyncItems.items || []
+)
+
+const { onResult: onProjectAccSyncItemsUpdated } = useSubscription(
+  onProjectAccSyncItemUpdatedSubscription,
+  () => ({
+    id: props.projectId
+  })
+)
+
+const disabledItemPredicate = (item: ProjectPageLatestItemsModelItemFragment) => {
+  return !!accSyncItems.value.find((i) => i.modelId === item.id)
+}
+
+onProjectAccSyncItemsUpdated((res) => {
+  // TODO ACC: Mutate local cache instead of refetch
+  refetchAccSyncItems()
+  triggerNotification({
+    type: ToastNotificationType.Info,
+    title: 'Acc Sync Item updated',
+    description: res.data?.projectAccSyncItemsUpdated.accSyncItem?.accFileLineageUrn
+  })
+})
+
+const fetchHubs = async () => {
```

Comment:
ideally, all of these fetches should use useFetch or useAsyncData for caching, but its probably fine for now